http-clients/src/test/java/com/palantir/remoting/http/OptionalAwareContractTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting.http;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableSet;
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+import com.squareup.okhttp.mockwebserver.RecordedRequest;
+import javax.net.ssl.SSLSocketFactory;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public final class OptionalAwareContractTest {
+
+    @Rule
+    public final MockWebServer server = new MockWebServer();
+
+    private FakeoInterface proxy;
+
+    @Before
+    public void before() {
+        proxy = FeignClients.standard().createProxy(Optional.<SSLSocketFactory>absent(),
+                ImmutableSet.of("http://localhost:" + server.getPort()),
+                FakeoInterface.class);
+        server.enqueue(new MockResponse().setBody("\"foo\""));
+    }
+
+    @Test
+    public void testAbsentPath() throws Exception {
+        proxy.path(Optional.<String>absent());
+        RecordedRequest takeRequest = server.takeRequest();
+        // note that this differs from feign's default behavior of leaving the template
+        // in the path. The default would be "/{opt}/foo"
```

Comment:
i think the `/null/foo` behaviour is actually preferred over `/{opt}/foo` since it's more obvious that an error is due to a null being passed in. for path parameters, the server has no way of disambiguating the `null` from `"null"` anyway.